components/plugin_marketplace/marketplace_item/marketplace_item.tsx

```diff
@@ -98,7 +98,18 @@ export default class MarketplaceItem extends React.PureComponent <MarketplaceIte
                     url={label.url}
                     color={label.color}
                 />
-            ),
+            ));
+        }
+
+        if (Constants.Integrations.FREEMIUM_USAGE_IGNORED_PLUGINS.includes(this.props.id)) {
+            labels.push(
+                <Label
+                    key={'CORE PLUGIN'}
+                    name={'CORE PLUGIN'}
+                    description={'This plugin does not count towards your limit of 5 enabled integrations'}
```

Comment:
I think this also needs to check if the limit applies, and optionally could also check if the user is on a cloud instance (optional because only cloud instances are going to have the limit applied). Since this class only defines the render method, it should be fairly easy to rewrite it as a functional compoment, which would allow using the `useGetLimits` hook and [`isCurrentLicenseCloud`](https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/selectors/entities/cloud.ts#L38) .

If we want the core plugin badge in non-cloud instances, then in that case I think we should have an alternate description.

Reply:
> If we want the core plugin badge in non-cloud instances, then in that case I think we should have an alternate description.

@neallred Yeah this is a good point. "Core Plugin" still has meaning outside of the Freemium context. 0/5 on what message we want to get across in this regard.

Reply:
does the description text need translation?

Reply:
@neallred I'm thinking we should use the `getCloudLimits` selector instead of `useGetLimits`, because these components are rendered a number of times on the same page, and the parent component is already fetching the limits.

Also, why check the Cloud license but not the Cloud Free feature flag?

Reply:
I don't think I said to not check the feature flag, although `useGetLimits` does have some feature flag checking built in. If you would rather use the selector, that sounds good to me; there is some chance that limits wouldn't already be loaded, and in that scenario it would do too many requests.